bp_common/test/src/demos/Makefile

```diff
@@ -1,16 +1,27 @@
 
 include Makefile.frag
 
-RISCV_GCC  = $(CROSS_COMPILE)gcc --static -nostartfiles -fPIC -march=rv64ima -mabi=lp64 -mcmodel=medany -I$(BP_TEST_DIR)/include
-RISCV_LINK = -static -nostartfiles -L$(BP_TEST_DIR)/lib -lperch -T src/riscv.ld
+RISCV_GCC       = $(CROSS_COMPILE)gcc
+RISCV_GCC_OPTS  = -march=rv64imafd -mabi=lp64 -mcmodel=medany -I $(BP_TEST_DIR)/include
+RISCV_LINK_OPTS = -T $(BP_TEST_DIR)/src/perch/riscv.ld -L$(BP_TEST_DIR)/lib -static -nostartfiles -lperch
+MKLFS           = dramfs_mklfs 128 64
+
 
 .PHONY: all
 
 all: $(addsuffix .riscv, $(BP_DEMOS))
 
 %.riscv:
-	$(RISCV_GCC) $(RISCV_LINK) -o $@ src/$*.c -lperch
+	$(RISCV_GCC) -o $@ $(wildcard src/$*/*.c) $(RISCV_GCC_OPTS) $(RISCV_LINK_OPTS)
+
+lfs.riscv: $(BP_TEST_DIR)/src/perch/start.S ./src/lfs/lfs.c
+	$(RISCV_GCC) -o $@ $^ $(wildcard src/lfs/*.c) -D_DRAMFS $(RISCV_GCC_OPTS) $(RISCV_LINK_OPTS)
+
+%/lfs.c:
```

Comment:
Maybe a makefile function that takes a filelist instead? We could add it to our Makefile.tests and then other suites could use it

 .txt is a little dangerous because it could grab README.txt or something like that